ui/lib/StreamStatus.tsx

```diff
@@ -1,36 +1,63 @@
-import * as React from "react";
+import React, { useState, useEffect } from "react";
 import { PieChart } from "react-minimal-pie-chart";
-
 import { Stream } from "@apeworx/apepay";
 
 export interface StreamStatusProps {
   stream: Stream;
+  chartType: "bar" | "pie";
 }
 
-const StreamStatus = (props: StreamStatusProps) => {
-  const [timeLeft, setTimeLeft] = React.useState(1); // using `1` to start w/ 100% left
-  React.useEffect(() => {
-    props.stream.timeLeft().then(setTimeLeft).catch(console.error);
-  }, [timeLeft]);
+const StreamStatus: React.FC<StreamStatusProps> = ({ stream, chartType }) => {
+  const [timeLeft, setTimeLeft] = useState<number>(1);
+  const [totalTime, setTotalTime] = useState<number>(1);
```

Comment:
might be better to use `null` as a guard to not trigger the interval until the data is fetched via RPC to initialize the component

then can display the component in a "loading..." state (using a spinner or something), to tell the user we are getting the information